File: app/src/main/java/com/softaai/designpatterns/java/behavioral/ChainOfResposibility/ConsoleLogger.java
package com.softaai.designpatterns.java.behavioral.ChainOfResposibility;

/**
 * Created by Amol Pawar on 05-04-2019.
 * softAai Apps
 */
public class ConsoleLogger extends AbstractLogger{

    public ConsoleLogger(int level){
        this.level = level;
    }

    @Override
    protected void write(String message) {
        System.out.println("Console::Logger: " + message);
    }
}

File: app/src/main/java/com/softaai/designpatterns/kotlin/behavioral/ConsoleLogger.kt
package com.softaai.designpatterns.kotlin.behavioral


/**
 * Created by Amol Pawar on 05-04-2019.
 * softAai Apps
 */
class ConsoleLogger(level : Int) : AbstractLogger() {

    init {
        this.level = level
    }

    override fun write(message: String) {
       println("Standard Console::Logger : " + message)
    }
}